File: Java/Chapter12/DigitalTv3d.java
public class DigitalTv3d extends DigitalTv {

    public DigitalTv3d(String region) {
	super(region); // 親クラスのコンストラクタ
    }

    public void display3d(int channel) {  
	if (_region.equals("新潟") ) {
	    switch (channel) {
	    case 8:
		System.out.println("NHK総合3D表示中...");
		break;
	    case 12:
		System.out.println("NHK教育3D表示中...");
		break;
	    default:
		System.out.println("3D表示できません...");
		break;
	    }
	}
	else
	    System.out.println("3D表示できません");
    }
}

File: Python/Chapter05/digital_tv_3d.py
from digital_tv import DigitalTv

class DigitalTv3d(DigitalTv):

    def __init__(self, region):
        # 基底クラスのコンストラクタ
        super().__init__(region)
        self.is3d = True

    def display(self, channel):
        if self.is3d == True:
            if self._region == '新潟':
                if channel == 8:
                    print('NHK総合3D表示中...')
                elif channel == 12:
                    print('NHK教育3D表示中...')
                else:
                    print('3D表示できません...')
            else:
                print('3D表示できません')
        else:
            super().display(channel)

if __name__ == '__main__':

    tv = DigitalTv3d('新潟')
    
    tv.display(8)
    tv.display(12)
    tv.display(1)
    
    tv.is3d = False
    
    tv.display(8)
    tv.display(12)
    tv.display(1)
